### DISCUSSION TITLE
Adding songs to playlist

### DISCUSSION THREAD

**Discussion by Arshad1-cmd** [MissingCore/Music#69] [Ideas]
Please add an option where you can choose multiple songs at once and then add them to a playlist instead of individually choosing them.

Also give more options under "Tracks" such as shuffle, recently added, added by date etc.

**Comment by The-Ladle** [comment]
Also if you could add albums too that would be so nice

**Reply by cyanChill** [reply]
Do you mean the sort option? If that's the case, then it'll probably be in the "Albums" tab with probably the options being: alphabetical and release year.

If you mean adding tracks to albums, I don't really see that being necessary since they should already be grouped by their metadata.

**Reply by The-Ladle** [reply]
I mean adding albums to playlists

**Reply by cyanChill** [reply]
Yea, that seems doable; no exact timeframe on when this will be implemented, but probably at the same time this feature.

I guess we can do a long-press on an album on the "Albums" tab and it'll do the multi-select idea. The other way of adding an album to a playlist can probably be an option on the specific album page itself.

**Comment by cyanChill** [comment]
I've mentioned in the [`Feature Road Map`](https://github.com/MissingCore/Music/discussions/9#discussioncomment-10131303) that I do plan on adding some sort of multi-select feature along with sorting.
- With sorting for tracks; the only options I can see we having is alphabetical & modification time (modification time is the only thing that's exposed to us via file metadata).
- With the shuffle & repeat buttons on the "Tracks" tab, I initially didn't want to put one there because I was initially thinking it would look a bit out a place with the design; now after doing the "Folders" tab since we've added a "breadcrumb", I think it could work out.

**Comment by cyanChill** [comment]
If you're using the APK version of the app, we released a preview of the next major version through the [`v2.0.0-rc.1`](https://github.com/MissingCore/Music/releases/tag/v2.0.0-rc.1) release, which improves how you add tracks to a playlist (new UI in the create/edit screen where you can search for tracks or albums and directly add them to the playlist), adds the shuffle/repeat/play/pause button set to the "Tracks" screen/tab, and adds sorting in alphabetical and last modified order for the "Tracks" screen/tab (which the player will respect when playing from that set of tracks given you're not playing the tracks in shuffled mode).
